tensornetwork/contractors/opt_einsum_paths/utils.py

```diff
@@ -42,7 +41,7 @@ def _get_path_nodes(nodes: Iterable[BaseNode], algorithm: Algorithm
   sorted_nodes = sorted(nodes, key=lambda n: n.signature)
 
   input_sets = [set(node.edges) for node in sorted_nodes]
-  output_set = get_all_edges(nodes) - get_all_nondangling(nodes)
+  output_set = get_subgraph_dangling(nodes)
```

Comment:
This seems to be a change in behavior.

Suppose we have two nodes A and B connected by a single edge and let nodes={A}. Then all edges is {(A, B)} and all non-dangling edges is also {(A, B)}, so before this PR, ouput_set would be empty. However, get_subgraph_dangling(nodes) will return the non-empty set {(A, B)} since the edge (A, B) is "relatively dangling".

Is this intentional?

Reply:
Yes, this is intentional and the point of the PR. So that edge between A and B would not be contracted, and is effectively treated as a "relatively dangling" edge.

This allows you to find optimal contractions of subnetworks to the tensor network. For example, in the [quantum supremacy Nasa paper](https://arxiv.org/pdf/1905.00444.pdf), they contract their network into 3 sections before combining those section together. With this PR, you can now pass only the nodes in say section D and it will find an optimal contraction path just for those nodes, leaving all of the nodes that are "relatively dangling" untouched.

Reply:
Right, makes sense. Thanks.